File: ros2_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class TestPublisher(Node):
    def __init__(self):
        super().__init__('test_publisher')
        self.publisher_ = self.create_publisher(String, 'test_topic', 10)
        timer_period = 1.0
        self.timer = self.create_timer(timer_period, self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = "Hello from Publisher!"
        self.get_logger().info(f'Publishing: {msg.data}')
        self.publisher_.publish(msg)  # ✅ 메시지 전송

def main(args=None):
    rclpy.init(args=args)
    node = TestPublisher()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()
    


 

if __name__ == '__main__':
    main()

File: ros2_publisher.js
const rclnodejs = require('rclnodejs');

async function main() {
  await rclnodejs.init();
  const node = rclnodejs.createNode('test_publisher');

  const publisher = node.createPublisher('std_msgs/msg/String', '/test_topic');

  setInterval(() => {
    const message = { data: "테스트 메시지: 아이스크림 주문 테스트 🚀" };
    console.log(`📤 메시지 발행: ${message.data}`);
    publisher.publish(message);
  }, 2000); // 2초마다 메시지 발행

  rclnodejs.spin(node);
}

main();
